cpp/src/binaryop/compiled/equality_ops.cu

```diff
@@ -27,33 +30,69 @@ void dispatch_equality_op(mutable_column_view& out,
 {
   CUDF_EXPECTS(op == binary_operator::EQUAL || op == binary_operator::NOT_EQUAL,
                "Unsupported operator for these types");
-  auto common_dtype = get_common_type(out.type(), lhs.type(), rhs.type());
-  auto outd         = mutable_column_device_view::create(out, stream);
-  auto lhsd         = column_device_view::create(lhs, stream);
-  auto rhsd         = column_device_view::create(rhs, stream);
-  if (common_dtype) {
-    if (op == binary_operator::EQUAL) {
-      for_each(stream,
-               out.size(),
-               binary_op_device_dispatcher<ops::Equal>{
-                 *common_dtype, *outd, *lhsd, *rhsd, is_lhs_scalar, is_rhs_scalar});
-    } else if (op == binary_operator::NOT_EQUAL) {
-      for_each(stream,
-               out.size(),
-               binary_op_device_dispatcher<ops::NotEqual>{
-                 *common_dtype, *outd, *lhsd, *rhsd, is_lhs_scalar, is_rhs_scalar});
-    }
+  if (is_struct(lhs.type()) && is_struct(rhs.type())) {
```

Comment:
This function is about 65 lines long. Could we separate out the struct logic into another function and call it here / return early, to reduce the long if/else branches in this function body?

```cpp
void dispatch_equality_op(...) {
  CUDF_EXPECTS(...);
  if (is_struct(lhs.type()) && is_struct(rhs.type())) {
    // Handle struct data
    dispatch_equality_op_structs(...);
    return;
  }
  // existing function body.
}
```